app/js/controllers/settings.js

```diff
@@ -5,21 +5,39 @@
 export default class SettingsController extends Controller {
   constructor(options) {
     super(options);
+
+    this.merging = false;
+
+    this._refreshAddons = this._refreshAddons.bind(this);
   }
 
   open() {
     this.view.modal.open();
 
-    AddonService.getAddons(window.location.host).then((addons) => {
-      this.view.setAddons(addons);
+    this._refreshAddons();
+
+    ['install', 'uninstall', 'enabledstatechange'].forEach((e) => {
```

Comment:
Seeing the 'install' hook here reminds me: do we have filtering to make sure that only Customizer-generated add-ons show up in the Customizer's add-ons list?